contract/lib.rs

```diff
@@ -0,0 +1,331 @@
+        points: u64,
+    }
+
+    #[ink(storage)]
+    pub struct Syndeo {
+        pub admin: AccountId,
+        pub members: Vec<AccountId>,
+        pub points_by_sender: Mapping<AccountId, u64>,
+        pub senders: Vec<AccountId>,
+        pub points_by_recipient: Mapping<AccountId, u64>,
+        pub recipients: Vec<AccountId>,
+        pub total_points: u64,
+        pub max_points_per_sender: u64,
+    }
+
+    impl Syndeo {
+        #[ink(constructor)]
+        pub fn new(max_points_per_sender: Option<u64>) -> Self {
+            let caller = Self::env().caller();
+            let mut members = Vec::new();
+            members.push(caller);
+
+            Self {
+                admin: caller,
+                members,
+                points_by_sender: Mapping::default(),
+                senders: Vec::new(),
+                points_by_recipient: Mapping::default(),
+                recipients: Vec::new(),
+                total_points: 0,
+                max_points_per_sender: max_points_per_sender.unwrap_or(10),
+            }
+        }
+
+        #[ink(message)]
+        pub fn set_admin(&mut self, new_admin: AccountId) -> Result<(), ContractError> {
+            self.check_admin()?;
+
+            if !self.is_member(&new_admin) {
+                self.add_member(new_admin)?;
+            }
+
+            self.admin = new_admin;
+
+            self.env().emit_event(AdminChanged { admin: new_admin });
+
+            Ok(())
+        }
+
+        #[ink(message)]
+        pub fn add_member(&mut self, new_member: AccountId) -> Result<(), ContractError> {
+            self.check_admin()?;
+
+            if self.members.contains(&new_member) {
+                return Err(ContractError::MemberAlreadyExists);
+            }
+
+            self.members.push(new_member);
+            self.env().emit_event(NewMember { member: new_member });
+
+            Ok(())
+        }
+
+        #[ink(message)]
+        pub fn remove_member(&mut self, member_to_remove: AccountId) -> Result<(), ContractError> {
+            self.check_admin()?;
+
+            match self.members.iter().position(|m| *m == member_to_remove) {
+                Some(member_index) => {
+                    self.members.remove(member_index);
+                    self.env().emit_event(MemberDeleted {
+                        member: member_to_remove,
+                    });
+                }
+                None => return Err(ContractError::MemberDoesNotExist),
+            };
+
+            Ok(())
+        }
+
+        #[ink(message)]
+        pub fn set_max_points_per_sender(
+            &mut self,
+            max_points_per_sender: u64,
+        ) -> Result<(), ContractError> {
+            self.check_admin()?;
+
+            if self.max_points_per_sender == 0 {
+                return Err(ContractError::MaxPointsPerSenderCannotBeZero);
+            }
+
+            self.max_points_per_sender = max_points_per_sender;
+
+            Ok(())
+        }
```

Comment:
Should we add a max value of points per sender? Something like 500

Reply:
The constructor has 10 as a default value.